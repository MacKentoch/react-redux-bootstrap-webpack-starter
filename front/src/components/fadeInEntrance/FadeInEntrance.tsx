import React, { ReactNode } from 'react';
import FadeInDiv from './styled/FadeInDiv';

// #region types
type Props = {
  children: ReactNode;
};
// #endregion

function FadeInEntrance({ children }: Props) {
  return <FadeInDiv startAnimation>{children}</FadeInDiv>;
}

FadeInEntrance.displayName = 'FadeInEntrance';

export default FadeInEntrance;
