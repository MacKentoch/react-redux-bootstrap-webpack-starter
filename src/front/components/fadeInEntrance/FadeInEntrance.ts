import React from 'react';
import FadeInDiv from './styled/FadeInDiv';

// #region types
type Props = {
  children: any,
};
// #endregion

function FadeInEntrance({ children }: Props) {
  return <FadeInDiv startAnimation>{children}</FadeInDiv>;
}

FadeInEntrance.displayName = 'FadeInEntrance';

export default FadeInEntrance;
