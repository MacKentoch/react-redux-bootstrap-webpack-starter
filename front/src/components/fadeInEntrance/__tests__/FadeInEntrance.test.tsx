import React from 'react';
import { render } from '@testing-library/react';
import FadeInEntrance from '../FadeInEntrance';

describe('FadeInEntrance component', () => {
  let rootElement: any = null;

  beforeEach(() => {
    rootElement = document.createElement('div');
    document.body.appendChild(rootElement);
  });

  afterEach(() => {
    rootElement && document.body.removeChild(rootElement);
    rootElement = null;
  });

  it('renders as expected', () => {
    const props = {};

    const { container } = render(
      <FadeInEntrance {...props}>
        <p>a child</p>
      </FadeInEntrance>,
      rootElement,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
