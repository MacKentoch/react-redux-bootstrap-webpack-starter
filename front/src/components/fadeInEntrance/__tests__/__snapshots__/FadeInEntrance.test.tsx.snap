// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FadeInEntrance component renders as expected 1`] = `
<div
  class="sc-bdVaJa jxVqOJ"
>
  <p>
    a child
  </p>
</div>
`;
